### ISSUE TITLE
Deno.bundle causes Assertion error in the TS compiler

### ISSUE THREAD

**Issue by littledivy** [denoland/deno#7518]
**Error**
![image](https://user-images.githubusercontent.com/34997667/93345909-0188bf80-f851-11ea-89dd-465e86890994.png)

**Steps to reproduce**

```typescript
const code: string = `
// foo.ts
export * from "./a.ts";
export * from "./b.ts";
`
await Deno.bundle(
        "/foo.ts",
        {
          "/foo.ts": code,
        },
        {
          target: "es3",
          module: "esnext",
        },
  );
```
where `a.ts` and `b.ts` can be any valid typescript module

**Deno** - 1.4.0
**v8** -  8.7.75
**typescript** - 4.0.2

**Comment by kitsonk** [comment]
I thought we had an issue for this, but we don't. The title should be deno bundle doesnt support export * exports.

**Comment by JordanMajd** [comment]
**TLDR:** I suspect `export * from "./a.ts"` counts as a redirect, but `buildLocalSourceFileCache` doesn't respect it.

`runtimeBundle` calls `buildLocalSourceFileCache(sourceFileMap)` but an assertion fails because the entries in `sourceFileMap` don't contain any `sourceCode`.

Removing the assertion removes the error, but the bundle still won't contain the expected exports from `foo.ts`.

While debugging, I found this could be overcome by replacing the call to `buildLocalSourceFileCache` with `buildSourceFileCache` and ensuring that the return's `output` was never empty `output: state.bundleOutput ? state.bundleOutput : {},`.

It looks like there are two differences between the functions:

1. `buildSourceFileCache` ensures that if the imported file is a redirect, the `mappedUrl` will reflect the redirect.
1. `buildLocalSourceFileCache` modifies the imports `mappedUrl` by removing `memmory://`.

After testing I found that 1. was the issue; `export * from "./a.ts";` counts as a redirect, but `buildLocalSourceFileCache` isn't updating the `mappedURL` to reflect that.

The solution was:

1. remove the offending `assert(entry.sourceCode.length > 0);` on line 493.
1. add the following right below line 505's `assert(importedFile);` to `buildLocalSourceFileCache`

```js
if (importedFile.redirect) {
  mappedUrl = importedFile.redirect;
}
```

Repro'd tested the fix with the following: `\deno.exe run --unstable --allow-all .\comp.ts`

```ts
//comp.ts
const code = `
// foo.ts
export * from "./a.ts";
export * from "./b.ts";
`;
await Deno.bundle(
  "/foo.ts",
  {
    "/foo.ts": code,
  },
  {
    target: "es3",
    module: "esnext",
  },
);
```
```ts
//a.ts
export default function(){
}
```
```ts
//b.ts
export default function(){
}
```

Incoming PR

**Comment by kitsonk** [comment]
For `deno bundle` this now works, but for `Deno.bundle()` it requires #8060 so we should keep this open until that.

**Comment by kitsonk** [comment]
#8060 is resolved... closing.
